pagopa/io-app
# feat: [IOPID-666] FL missing pin logout flow

## Short description
This PR adds the logout flow when the pin is missing during a token refresh

## List of changes proposed in this pull request
- ts/features/fastLogin/saga/tokenRefreshSaga.ts: added `isPinAvailable` check
- ts/features/fastLogin/screens/FastLoginModals.tsx: refactored fast login modals and added a new modal for `no-pin-error`
- [ts/features/fastLogin/store/actions/index.ts, ts/features/fastLogin/store/reducers/index.ts]: added new action state logic
- ts/sagas/__tests__/initializeApplicationSaga.test.ts: fixed tests
- ts/sagas/startup.ts: moved `watchLogoutSaga` to the top (otherwise dispatching the logout action during onboarding would have no effect)
- ts/screens/modal/RootModal.tsx: refactored

## How to test
Enable fast login on dev server and run it.
Let the token expire before setting the pin.
If you are in app, you should see a new modal that informs you and you should be logged out.
If you reopen the app, while in onboarding (before pin setting), you should be logged out.
